packages/lib-classifier/src/plugins/drawingTools/models/tools/Tool/Tool.js

```diff
@@ -66,6 +66,10 @@ const Tool = types.model('Tool', {
       self.marks.delete(mark.id)
     }
 
+    function handlePointerDown (mark) {
+      return self.createMark(mark)
```

Comment:
I'm thinking this is still needed for something like a polygon tool using a touch screen. If using a polygon tool, a second mark placed will probably hit the 
`activeTool.handlePointerDown && activeTool.handlePointerDown(convertEvent(event))` line, which will fail if that function is not defined on the tool.

Reply:
That line checks if `activeTool.handlePointerDown` exists, so won’t fail if it isn’t defined.

Reply:
But will it still add a mark for something like a polygon if using a touch screen? I believe using a touchscreen would call `onPointerDown` in the `InteractionLayer` with each touch, which would essentially do nothing for a polygon without this fallback.

Reply:
`createMark()` starts a new Mark, so it would place the first point for a new Polygon. TranscriptionLine is a special case of Polygon with only two points, I suppose.

Reply:
Yeah, I think we need to pick a single code path here:

- Option 1: The `InteractionLayer` does an existential check for the existence of the `handlePointerDown` action before it calls it. If it exists on the current active tool and the mark is `creating`, then it is called and the `onPointerDown` function ends with returning true. If it does not exist and the mark is not `creating`, `createMark` is called by the same function in the `InteractionLayer` and then sets the new mark as active, its initial position, sets the subtask visibility and returns false.
- Option 2: The `InteractionLayer` does not do an existential check and we have the generic `handlePointerDown` action defined on the generic `Tool` model that all tools are composed with. The generic version of this handler calls createMark and then handles any other actions needing to be done like setting the active mark, etc.

As it is now, the code path can be a little hard to follow since there's two paths for how a mark might get created and it's not clear what the conditions are for when one happens vs the other (an active tool's `handlePointerDown` creates the mark vs the `onPointerDown` in `InteractionLayer`). Is there a chance `createMark` gets called twice when we don't want it to? I'm not sure. 

This further refactoring, though, I think is out of scope of this fix, but I'd like to see it at some point.

Reply:
Just a note that returning true from an event handler historically signals that the browser should continue to handle the event with its default handler. So, I think we want to return false except in the default case, where the interaction layer ignores the event. 

I don't think any modern browser uses Boolean returns from a handler, but preventing the default action and also returning true can be confusing to read.

Reply:
> Is there a chance createMark gets called twice when we don't want it to? I'm not sure.

I think it does. My understanding is that this handler only gets called when `creating` is true, so we would have mark creation happening twice ie. creating two new ellipses or points or lines when we only want one. In actuality, `pointerUp` cancels creation for each of those shapes so this handler is never called. It will be called when we build tools that don't end creation on pointerUp eg. polygons or maybe freehand drawing. Ellipses probably shouldn't end creation on pointerUp either. I think that was a mistake on my part.

If there isn't a test already, I think there should be a test that pointerDown on the interaction layer does not create a new mark when `creating` is true.

Reply:
Following up on @srallen's comments about changing the code paths here, which is confusing, any change to the drawing API should also be documented in the README for the drawing tools.